packages/analytics-js-integrations/src/utils/commonUtils.js

```diff
@@ -1,17 +1,20 @@
-import _isEmpty from 'lodash.isempty';
-import _toString from 'lodash.tostring';
-import { isDefined } from '@rudderstack/analytics-js-common/utilities/checks';
-import { pick as ramdaPick, pickBy } from 'ramda';
+import { isDefined, isString } from '@rudderstack/analytics-js-common/utilities/checks';
+import { pick as ramdaPick, pickBy, isEmpty } from 'ramda';
 
-const isNotEmpty = x => !_isEmpty(x);
+const isNotEmpty = x => {
+  if (typeof x === 'number' || typeof x === 'boolean') {
+    return isEmpty(x); // Numbers and booleans are inherently "not empty"
+  }
+  return !isEmpty(x);
+};
 const isNotNull = x => x != null;
 const isDefinedAndNotNull = x => isDefined(x) && isNotNull(x);
 const isDefinedAndNotNullAndNotEmpty = x => isDefined(x) && isNotNull(x) && isNotEmpty(x);
 const removeUndefinedValues = obj => pickBy(isDefined, obj);
 const removeNullValues = obj => pickBy(isNotNull, obj);
 const removeUndefinedAndNullValues = obj => pickBy(isDefinedAndNotNull, obj);
 const removeUndefinedAndNullAndEmptyValues = obj => pickBy(isDefinedAndNotNullAndNotEmpty, obj);
-const isBlank = value => _isEmpty(_toString(value).trim());
+const isBlank = value => isEmpty(String(value).trim());
```

Comment:
Instead of converting to a string, check if it is a string and return false otherwise.

Reply:
I have tested and they were working in the same way. I have added test cases as well on the adobeanalytics/util.test.js file. It is triggered for the `handleList` function.